app/mailers/notification_mailer.rb

```diff
@@ -103,11 +103,14 @@ def new_reply(reply, user)
     @reply = reply
     @user = user
     return if EmailAddress.pluck(:email).include?(user.email.to_s)
-    
+
     displayed_to_field = reply.notified_users.where(agent: false).pluck(:email)
     displayed_to_field = user.email if displayed_to_field.empty?
-    mail(smtp_envelope_to: user.email, to: displayed_to_field,
+
+    message = mail(smtp_envelope_to: user.email, to: displayed_to_field,
```

Comment:
This sets an `smtp_envelope_to` header on the message itself. You don't need this.